### ISSUE TITLE
Provide JSON output for list commands

### ISSUE THREAD

**Issue by peterj** [fnproject/cli#375]
Having JSON output makes it much easier to parse/script things; the current output that's especially hard to parse is the output of the `fn list calls` command.

**Comment by peterj** [comment]
Looking at this, it might be useful if this is enabled on other resources as well (apps, routes, ...)

**Comment by denismakogon** [comment]
some of these already provide JSON objects, take a look at `fn inspect`

**Comment by peterj** [comment]
True, but I was thinking more in this way: 
```
$ fn list apps --output json
[
    {
        "name": "app1"
    },
    {
        "name": "app2"
    }
]
```

Or listing the calls:
```
$ fn list calls app1 --json
[
    {
        "id": "01CMBC2R72R2M0250ZJ0000003",
        "appId": "01CM8P0Y60R2M023RZJ0000001",
        "route": "/testapp",
        "createdAt": "2018-08-07T16:38:16.674-07:00",
        "startedAt": "2018-08-07T16:38:16.674-07:00",
        "completedAt": "2018-08-07T16:38:16.686-07:00",
        "status": "success"
    },
    {
        "id": "01CMBC2R72R2M0250ZJ0000003",
        "appId": "01CM8P0Y60R2M023RZJ0000001",
        "route": "/testapp",
        "createdAt": "2018-08-07T16:38:16.674-07:00",
        "startedAt": "2018-08-07T16:38:16.674-07:00",
        "completedAt": "2018-08-07T16:38:16.686-07:00",
        "status": "success"
    }
]
```

This could make any automation scenarios much easier - for example you could pipe the output of an Fn command to a different binary (jq to parse, or something else). The way we output stuff today (calls in particular) is not parsable at all.

**Comment by shaunsmith** [comment]
+1 I thought I'd filed this enhancement request!  We definitely need optional JSON output for scripting support.

**Comment by peterj** [comment]
This was merged yesterday - #382

**Comment by shaunsmith** [comment]
Thanks @peterj !
